vignettes/tcplfit2-vignette.Rmd

```diff
@@ -232,89 +269,95 @@ dat[wllt == "t" & acid %in% nconc[nconc > 1, acid],
+
+```{r example3_mthdlist, fig.height = 6, fig.width = 7, warning = FALSE}
+## Check out all level-3 methods for normalization in tcpl
 DT::datatable(tcpl::tcplMthdList(3),rownames= FALSE, options = list(scrollX = T))
 ```
 
-### Normalization
-Here three normalization methods are selected and applied to the data.  Note because of the way `tcpl` handles the application of functions, the dataframe must be called `dat`. In the future, `tcpl` will export these functions so that they can be applied to any dataset without the need for a specific name or dummy aeid.
+```{r example3 normalize}
+# calculate bval
+dat[, bval := median(cval[wllt == "n"]), by = list(apid)]
+# calculate pval
+dat[, pval := median(cval[wllt %in% c("m","o")], na.rm = TRUE), by = list(apid, wllt, conc)]
+dat[, pval := min(pval, na.rm = TRUE), by = list(apid)]
 
-```{r example4_mc3methods, fig.height = 6, fig.width = 7, results = 'hide'}
-# apply level 3 methods
-## These methods directly apply the normalization methods from tcpl without the need for a DB connection
-lapply(mthd_funcs[["bval.apid.nwlls.med"]](dummy_aeid), eval)
-lapply(mthd_funcs[["pval.apid.medncbyconc.min"]](dummy_aeid),eval)
-lapply(mthd_funcs[["resp.pc"]](dummy_aeid),eval)
+# Calculate normalized responses
+dat[, resp := ((cval - bval)/(pval - bval) * 100)]
 ```
 
-Level 4 determines the baseline variability, or noise, that will later be used for cutoff calculation. Using the established concentration index, the level 4 methods can be loaded in a similar way to level 3.
+Before model fitting, we need to determine the median absolute deviation around baseline (`BMAD`) and baseline variability (`onesd`), which are later used for cutoff and benchmark response (`BMR`) calculations, respectively. This is part of level-4 processing in `tcpl`. `tcplfit2` assumes no effect occurs in the two lowest concentrations. Thus, we consider test wells in the two lowest concentrations as our baseline to calculate `BMAD` and `onesd`.
 
-```{r example4_mthdlist_l4, fig.height = 6, fig.width = 7}
-mthd_funcs_l4 <- tcpl:::mc4_mthds()
+`BMD` can be calculated as the median absolute deviation of the data in control wells too. Check out other methods of determining `BMAD` and `onesd` used in `tcpl`.
+
+```{r example3_mthdlist_l4, fig.height = 6, fig.width = 7}
+# Check out all level-4 methods for normalization in tcpl
 DT::datatable(tcpl::tcplMthdList(4), rownames= FALSE, options = list(scrollX = T))
 ```
 
-There are much fewer level 4 methods, but generally it is a requirement to assign a method that calculates the bmad and assign a method that calculates the standard deviation of the noise for `tcplfit2` fitting.
+`BMAD` and `onesd` should be calculated on the assay level. The example data set only contains one assay compound, so here we only need to calculate once.
```

Comment:
Please clarify/flesh-out description here. If I am a naieve reader of this vignette this may somewhat confusing.